tooling-mp-jwt-auth-tool/src/main/java/org/jboss/eap/qe/microprofile/jwt/auth/tool/JwsHelper.java

```diff
@@ -0,0 +1,192 @@
+package org.jboss.eap.qe.microprofile.jwt.auth.tool;
+
+import java.net.URI;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.util.List;
+import java.util.Map;
+
+import org.jose4j.jwa.AlgorithmConstraints;
+import org.jose4j.jws.AlgorithmIdentifiers;
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.jwt.consumer.JwtConsumer;
+import org.jose4j.jwt.consumer.JwtConsumerBuilder;
+import org.jose4j.jwt.consumer.JwtContext;
+
+/**
+ * Utility wrapper for the jose4j library for creating and reading JWS (JWT Signed Tokens)
+ * 
+ * @author tborgato
+ */
+public class JwsHelper {
+
+    private RsaKeyHelper rsaHelper;
+    private EcKeyHelper ecHelper;
+
+    private RsaKeyHelper getRsaHelper() {
+        if (rsaHelper == null)
+            rsaHelper = new RsaKeyHelper();
+        return rsaHelper;
+    }
+
+    private EcKeyHelper getEcHelper() {
+        if (ecHelper == null)
+            ecHelper = new EcKeyHelper();
+        return ecHelper;
+    }
+
+    /**
+     * Generates a signed JWT;
+     *
+     * @param algorithm used to sign the JWT
+     * @param privateKey used to sign the JWT
+     * @param kid key ID Header Parameter
+     * @param issuer JWT claim
+     * @param audience JWT claim
+     * @param subject JWT claim
+     * @param groups JWT claim
+     * @param customClaims custom JWT claims
+     * @return {@link JsonWebSignature} representing the JWS
+     * @throws Exception in case anything goes wrong while generating the JWS
+     */
+    public JsonWebSignature generateProperSignedJwt(
```

Comment:
A builder pattern would better option here.

Reply:
@honza-kasik good idea, do you want me to modify it in this PR?

Reply:
@tommaso-borgato This can be done in next one, to keep this "short".